tests/integration/adapter-tests.js

```diff
@@ -99,4 +99,30 @@ export default function adapterTests() {
     expect(await persister.find(recordingId)).to.be.null;
     expect(responseCalled).to.be.true;
   });
+
+  it('should have responses after flushing', async function() {
+    const { server } = this.polly;
+    const requests = [];
+
+    server.get(this.recordUrl()).intercept(async (req, res) => {
+      requests.push(req);
+      await server.timeout(5 * requests.length);
+      res.sendStatus(200);
+    });
+
+    this.fetchRecord();
+    this.fetchRecord();
+    this.fetchRecord();
+
+    // Wait for all requests to be made
+    // (Puppeteer adapter takes some time to make the actual request)
+    while (requests.length !== 3) {
```

Comment:
Is this only noticed within the puppeteer adapter?  Worry we could mask potential future bugs by doing this always in all the adapter tests.

Reply:
> Is this only noticed within the puppeteer adapter?

Yes.

We use these tests in many places for both node and browser. I'm not really sure whats a better way to do this while making sure all adapters are tested for.

Reply:
> I wonder also if this could be mitigated by resolving the flush promise in the next frame.

What do you mean by this?

Reply:
> We use these tests in many places for both node and browser. I'm not really sure whats a better way to do this while making sure all adapters are tested for.

Skip the test in puppeteer.  Then create a test with this while loop within puppeteer's test suite so that we don't mask future bugs on the other adapters. 

> I wonder also if this could be mitigated by resolving the flush promise in the next frame.

Lets chat, I'd like to know more about what is going on here.  Specifically "takes some time".  Do we know how much time?  Does this while loop loop more than once?